fetch.bs

```diff
@@ -2349,38 +2348,59 @@ steps:
    <a>connection</a>.
   </ol>
 
+ <li>
+  <p>Let <var>proxies</var> be the result of performing proxy resolution for <var>url</var> in an
+  <a>implementation-defined</a> manner. If a proxy preference order has not been explicitly
+  configured, then SOCKS proxies should be preferred to HTTPS proxies which should be preferred to
```

Comment:
This part is WebSocket-specific.

For HTTP and HTTPS we prefer to match the proxy type.

The reasoning is that the proxy configured for "HTTPS" may be optimised for the "CONNECT" method (ie. low latency, no cache) and the proxy configured for "HTTP" may be optimised oppositely. Which protocol they use is orthogonal to what protocol they are configured for. In fact, it might be perfectly reasonable to use `https://proxy.local/` for the HTTP proxy and `http://proxy.local:3128/` for the HTTPS proxy.

SOCKS works like a fallback in the non-WebSocket case, and will only be used if a matching proxy type is not there.

This only applies to static configurations. If proxy configuration happens via a JavaScript file (`proxy.pac` or `wpad.dat`), it decides the priority.

The way that manual proxy configuration works varies by OS, and we probably shouldn't say anything about the priority order except in the WebSocket case where the underlying RFC6455 attempts to dictate it.